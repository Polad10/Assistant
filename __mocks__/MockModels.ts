const appointment_1 = { id: 1, datetime: '2024-01-01 19:44:00.000Z', actions: null, treatment_id: 1 }
const appointment_2 = { id: 2, datetime: '2024-01-02 19:44:00.000Z', actions: null, treatment_id: 2 }

const patient_1 = {
  id: 1,
  first_name: 'FirstName',
  last_name: 'LastName',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const patient_2 = {
  id: 2,
  first_name: 'FirstName2',
  last_name: 'LastName2',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const payment_1 = {
  id: 1,
  date: '2024-01-01 19:44:00.000Z',
  amount: 100,
  treatment_id: 1,
}

const payment_2 = {
  id: 2,
  date: '2024-01-02 19:44:00.000Z',
  amount: 100,
  treatment_id: 2,
}

const treatment_1 = {
  id: 1,
  start_date: '2024-01-01 19:44:00.000Z',
  end_date: null,
  title: 'Title',
  patient_id: 1,
  price: 100,
}

const treatment_2 = {
  id: 2,
  start_date: '2024-01-02 19:44:00.000Z',
  end_date: null,
  title: 'Title2',
  patient_id: 2,
  price: 100,
}

const setting = {
  language: 'en',
}

export const mockModels = {
  appointment_1,
  appointment_2,
  patient_1,
  patient_2,
  payment_1,
  payment_2,
  treatment_1,
  treatment_2,
  setting,
}
