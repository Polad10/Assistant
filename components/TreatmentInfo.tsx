import { View, Text, StyleSheet } from 'react-native'
import MainView from './MainView'
import { Icon, LinearProgress } from '@rneui/themed'
import { DateTime } from 'luxon'
import { Treatment } from '../modals/Treatment'
import { Colors } from '../types/Colors'
import { useTheme } from '@react-navigation/native'
import { useContext } from 'react'
import { DataContext } from '../contexts/DataContext'
import { getTotalPayment } from '../helpers/PaymentHelper'

type Props = {
  treatment: Treatment
}

export default function TreatmentInfo(props: Props) {
  const { colors } = useTheme()
  const context = useContext(DataContext)!

  const treatment = props.treatment

  const payments = context.payments?.filter((p) => p.treatment_id === treatment?.id) ?? []
  const totalPayment = getTotalPayment(payments)

  return (
    <MainView>
      <View style={[styles(colors).card, styles(colors).rounded, styles(colors).priceView]}>
        <Icon
          containerStyle={{
            marginRight: 20,
          }}
          name='wallet-outline'
          type='ionicon'
          size={25}
          color={colors.notification}
          reverse
        />
        <View style={{ flex: 1 }}>
          <View style={styles(colors).priceDetailsView}>
            <View>
              <Text style={styles(colors).text}>Price</Text>
              <Text style={[styles(colors).paymentsNrText, styles(colors).halfOpacity]}>
                {payments.length} Payments
              </Text>
            </View>
            <View style={{ flexDirection: 'row' }}>
              <Text style={[styles(colors).valueText, styles(colors).bold]}>{totalPayment} ₼ </Text>
              <Text style={[styles(colors).valueText, styles(colors).halfOpacity]}> of </Text>
              <Text style={[styles(colors).valueText, styles(colors).bold]}> {treatment.price} ₼</Text>
            </View>
          </View>
          <LinearProgress
            style={{ marginTop: 10 }}
            value={totalPayment / treatment.price}
            color={colors.notification}
          />
        </View>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <View style={[styles(colors).startDateView, styles(colors).card, styles(colors).rounded]}>
          <Text style={styles(colors).text}>Start</Text>
          <Icon
            name='hourglass-start'
            type='font-awesome'
            color='orange'
            size={25}
            containerStyle={styles(colors).dateIconContainer}
          />
          <Text style={[styles(colors).valueText, styles(colors).bold]}>
            {DateTime.fromISO(treatment.start_date).toFormat('MMMM d, yyyy')}
          </Text>
        </View>
        <View style={[styles(colors).endDateView, styles(colors).card, styles(colors).rounded]}>
          <Text style={styles(colors).text}>End</Text>
          <Icon
            name='hourglass-end'
            type='font-awesome'
            color='lightgreen'
            size={25}
            containerStyle={styles(colors).dateIconContainer}
          />
          <Text style={[styles(colors).valueText, styles(colors).bold]}>
            {treatment.end_date ? DateTime.fromISO(treatment.end_date).toFormat('MMMM d, yyyy') : '-'}
          </Text>
        </View>
      </View>
    </MainView>
  )
}

const styles = (colors: Colors) =>
  StyleSheet.create({
    card: {
      backgroundColor: colors.card,
      padding: 20,
      marginTop: 5,
    },
    rounded: {
      borderRadius: 10,
    },
    priceView: {
      flexDirection: 'row',
      alignItems: 'center',
      marginTop: 0,
    },
    priceDetailsView: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
    text: {
      color: colors.text,
      fontSize: 20,
    },
    paymentsNrText: {
      color: colors.text,
      fontSize: 16,
      marginTop: 5,
    },
    halfOpacity: {
      opacity: 0.5,
      fontWeight: '200',
    },
    bold: {
      fontWeight: 'bold',
    },
    valueText: {
      color: colors.text,
      fontSize: 18,
    },
    startDateView: {
      flex: 1,
      marginRight: 2.5,
    },
    endDateView: {
      flex: 1,
      marginLeft: 2.5,
    },
    dateIconContainer: {
      alignItems: 'flex-start',
      marginTop: 25,
      marginBottom: 15,
    },
  })
