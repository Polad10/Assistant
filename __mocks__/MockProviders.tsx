import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
