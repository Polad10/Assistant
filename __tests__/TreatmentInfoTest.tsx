import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import TreatmentInfo from '../components/TreatmentInfo'
import { mockModels } from '../__mocks__/MockModels'

describe('TreatmentInfo', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <TreatmentInfo treatment={mockModels.treatment_1} />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
