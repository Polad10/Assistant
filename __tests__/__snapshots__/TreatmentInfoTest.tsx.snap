// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreatmentInfo renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#1a1a1a",
          "marginTop": 5,
          "padding": 20,
        },
        {
          "borderRadius": 10,
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "marginRight": 20,
          "overflow": "hidden",
        }
      }
      testID="RNE__ICON__CONTAINER"
    >
      <View
        accessibilityRole="button"
        android_ripple={
          {
            "borderless": false,
            "color": "rgba(0, 0, 0, 0.3)",
            "radius": -5,
          }
        }
        disabled={false}
        testID="RNE__ICON__CONTAINER_ACTION"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          testID="RNE__ICON"
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "#faf8ff",
                "fontSize": 20,
              }
            }
          >
            Price
          </Text>
          <Text
            style={
              [
                {
                  "color": "#faf8ff",
                  "fontSize": 16,
                  "marginTop": 5,
                },
                {
                  "fontWeight": "200",
                  "opacity": 0.5,
                },
              ]
            }
          >
            1
             
            Payments
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#faf8ff",
                  "fontSize": 18,
                },
                {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            100
             ₼ 
          </Text>
          <Text
            style={
              [
                {
                  "color": "#faf8ff",
                  "fontSize": 18,
                },
                {
                  "fontWeight": "200",
                  "opacity": 0.5,
                },
              ]
            }
          >
             
            of
             
          </Text>
          <Text
            style={
              [
                {
                  "color": "#faf8ff",
                  "fontSize": 18,
                },
                {
                  "fontWeight": "bold",
                },
              ]
            }
          >
             
            100
             ₼
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="progressbar"
        accessibilityValue={
          {
            "max": 1,
            "min": 0,
            "now": 1,
          }
        }
        accessible={true}
        onLayout={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(255, 0, 255, 0.4)",
              "height": 4,
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            },
            {
              "marginTop": 10,
            },
          ]
        }
        testID="RNE__LinearProgress_Bar"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgb(255, 0, 255)",
              "flex": 1,
              "transform": [
                {
                  "translateX": -0,
                },
                {
                  "scaleX": 0.0001,
                },
              ],
            }
          }
          testID="RNE__LinearProgress_Progress"
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
            "marginRight": 2.5,
          },
          {
            "backgroundColor": "#1a1a1a",
            "marginTop": 5,
            "padding": 20,
          },
          {
            "borderRadius": 10,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#faf8ff",
            "fontSize": 20,
          }
        }
      >
        Start
      </Text>
      <View
        style={
          {
            "alignItems": "flex-start",
            "marginBottom": 15,
            "marginTop": 25,
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text />
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#faf8ff",
              "fontSize": 18,
            },
            {
              "fontWeight": "bold",
            },
          ]
        }
      >
        Invalid DateTime
      </Text>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "marginLeft": 2.5,
          },
          {
            "backgroundColor": "#1a1a1a",
            "marginTop": 5,
            "padding": 20,
          },
          {
            "borderRadius": 10,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#faf8ff",
            "fontSize": 20,
          }
        }
      >
        End
      </Text>
      <View
        style={
          {
            "alignItems": "flex-start",
            "marginBottom": 15,
            "marginTop": 25,
            "overflow": "hidden",
          }
        }
        testID="RNE__ICON__CONTAINER"
      >
        <View
          accessibilityRole="button"
          android_ripple={
            {
              "borderless": false,
              "color": "rgba(0, 0, 0, 0.3)",
              "radius": -5,
            }
          }
          disabled={false}
          testID="RNE__ICON__CONTAINER_ACTION"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            testID="RNE__ICON"
          >
            <Text />
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#faf8ff",
              "fontSize": 18,
            },
            {
              "fontWeight": "bold",
            },
          ]
        }
      >
        -
      </Text>
    </View>
  </View>
</View>
`;
